CMakeLists.txt

```diff
@@ -62,6 +62,15 @@ endif()
 # ----- HPX
 find_package(HPX REQUIRED)
 
+# In HPX 1.3.0 there is compiler matching check, but it does not check matching of build types.
+# Since it is source of headaches, waiting for HPX to add this check, we check on our own
+if (NOT "${HPX_BUILD_TYPE}" STREQUAL "${CMAKE_BUILD_TYPE}")
+  message(WARNING
+  "CMAKE_BUILD_TYPE does not match HPX_BUILD_TYPE=${HPX_BUILD_TYPE}\n"
+  "This project uses ${CMAKE_BUILD_TYPE}, while HPX uses ${HPX_BUILD_TYPE}.\n"
+  "Please add -DCMAKE_BUILD_TYPE=${HPX_BUILD_TYPE} to your cmake command or use a debug build of HPX")
```

Comment:
This message is a bit strange. HPX_BUILD_TYPE is repeated multiple times.
Moreover using Release with a Debug HPX, it suggests to use a Debug HPX.
I suggest:
"The build type used (CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}) does not match with HPX build type (HPX_BUILD_TYPE=${HPX_BUILD_TYPE})."
"Please either change CMAKE_BUILD_TYPE to ${HPX_BUILD_TYPE} or use a ${CMAKE_BUILD_TYPE} build of HPX"